### PR TITLE
Redirect handling.

### LINKED ISSUE #17: Redirect handling
Werkzeug can raise a `RequestRedirect` if eg. view is routed to `/foo/`, and a GET request is made to `/foo` (without the trailing slash).

We ought to be dealing with those in our [exception handler function](https://github.com/tomchristie/apistar/blob/master/apistar/routing.py#L139), and returning an appropriate response.

### PR DESCRIPTION
**Pull request by kinabalu**
Return HTTP 302 responses when we get a redirect from Werkzeug.

Closes #17

### CODE REVIEW

File: apistar/routing.py [hunk 1]
```diff
@@ -138,11 +138,17 @@ def lookup(self, path: str, method: str) -> RouterLookup:
             raise exceptions.NotFound()
         except werkzeug.exceptions.MethodNotAllowed:
             raise exceptions.MethodNotAllowed()
+        except werkzeug.routing.RequestRedirect:
+            raise exceptions.Found(path + '/')
```
Comment (tomchristie) [inline apistar/routing.py]: Does `RequestRedirect` include something like a url property already?
Can we do, something like this maybe?...

    except werkzeug.routing.RequestRedirect as exc:
        raise exceptions.Found(exc.url)

File: tests/test_routing.py [hunk 2]
```diff
@@ -89,6 +89,14 @@ def test_405():
     }
 
 
+def test_found_no_slash():
+    response = client.get('/found')
+    assert response.status_code == 200
```
Comment (tomchristie) [inline tests/test_routing.py]: It'd probably make more sense for us to check the result of `client.get('/found', follow_redirects=False)`, and ensure that a 302 is returned, with the correct headers.

**Comment by tomchristie** [comment]
(updated title)

Also, apologies looks like I resolved a merge conflict before you were done working on the PR. GitHub just made it too nice and easy! 😄

**Comment by kinabalu** [comment]
heh, easy enough to fix the conflict, just like we likes it

**Comment by tomchristie** [comment]
Ace, thanks!

**Pull request merged**